.github/workflows/ci.yml

```diff
@@ -65,34 +67,24 @@ jobs:
       - name: Install dependencies
         uses: ramsey/composer-install@v2
 
-      - name: Load php-cs-fixer cache
-        id: php-cs-fixer-cache
-        uses: actions/cache/restore@v3
+      - name: Cache PHP-CS-Fixer
+        uses: actions/cache@v3
         with:
-          path: .php-cs-fixer.cache
-          key: php-cs-fixer-${{ github.ref_name }}
-          restore-keys: |
-            php-cs-fixer-
+          path: ${{ env.PHP_CS_FIXER_CACHE_DIR }}
+          key: php-cs-fixer-${{ runner.os }}-${{ steps.setup_php.outputs.php-version }}
 
       - name: Check code style
         run: composer run cs
 
-      - name: Push php-cs-fixer branch cache
-        id: push-php-cs-fixer-cache-branch
-        if: always()
```

Comment:
The reason the cache step was split up into the restore and the push parts, was because of this always() key. If there is a php error, the codestyle step will fail but there is still work done that we can save for the next run that fixes the PHP error. Now, whenever there is a PHP error, the job will take a long time, then the PR author will push the fix and the workflow will take a long time again. AFAIK there is no way to always update the cache when one job after it fails.

Reply:
> Now, whenever there is a PHP error, the job will take a long time

No :)
The previous content of the cache will be used.

Reply:
Yes, which together with the comment chain below might be very outdated. If you push 20 times and a PHP file is broken on the first push, we do the work anyway, so why not update our cache with the files we can cache and have a faster workflow run when we do fix the broken file?

Reply:
All right.
https://github.com/actions/cache/blob/main/caching-strategies.md#saving-cache-even-if-the-build-fails

Reply:
Done.